homeassistant/components/thread/discovery.py

```diff
@@ -31,6 +34,65 @@ class ThreadRouterDiscoveryData:
     network_name: str | None
     server: str | None
     vendor_name: str | None
+    addresses: list[str] | None
+    thread_version: str | None
+
+
+def async_discovery_data_from_service(
+    service: AsyncServiceInfo,
+) -> ThreadRouterDiscoveryData:
+    """Get a ThreadRouterDiscoveryData from an AsyncServiceInfo."""
+
+    def try_decode(value: bytes | None) -> str | None:
+        """Try decoding UTF-8."""
+        if value is None:
+            return None
+        try:
+            return value.decode()
+        except UnicodeDecodeError:
+            return None
```

Comment:
Should this base64 encode the value prefixed with `encode-error:`, so we know it's not None?

Reply:
ok let's do it in another PR then.

Reply:
I guess I would like that detail in the diagnostic report, but the same decoder is used by the thread panel and It would be weird there. Probably better to show nothing that show the user garbage? Maybe leave as is for now and we can revisit if we see it in practice? Can always include the raw `TXT` record if we start seeing trash border routers?